sql/procedures.mysql

```diff
@@ -122,6 +122,55 @@ CREATE OR REPLACE VIEW beekeeper_activity AS
     ON b.beekeeper_id = lm.beekeeper_id
     GROUP BY b.beekeeper_id;
 
+-- show semaphores by job with logic name
+--
+-- Usage:
+--       select * from job_semaphores;
+--       select * from job_semaphores where controlling_logic_name = 'an_analysis';
+--
+--       -- get semaphore counts for jobs of a particular semaphored analysis
+--       select distinct semaphored_job_id, semaphore_count from job_semaphores
+--       where semaphored_logic_name = 'an_analysis'
+
+CREATE OR REPLACE VIEW job_semaphores AS
+    SELECT c_ab.logic_name AS controlling_logic_name, s_ab.logic_name as semaphored_logic_name,
+           c_j.job_id as controlling_job_id,
+           (s.local_jobs_counter + s.remote_jobs_counter) AS semaphore_count,
+           s.dependent_job_id AS semaphored_job_id, s.dependent_semaphore_url
+    FROM semaphore s
+    LEFT JOIN job c_j ON s.semaphore_id = c_j.controlled_semaphore_id
+    LEFT JOIN analysis_base c_ab ON c_ab.analysis_id = c_j.analysis_id
+    LEFT JOIN job s_j ON s_j.job_id = s.dependent_job_id
+    LEFT JOIN analysis_base s_ab ON s_ab.analysis_id = s_j.analysis_id;
+
+-- show a summary of semaphored jobs and/or dependent semaphore URLs
+--
+-- Usage:
+--       select * from semaphore_sumary;
+--       select * from semaphore_summary where semaphored_logic_name like '%nalysi%';
+
+CREATE OR REPLACE VIEW semaphore_summary AS
+   SELECT s.semaphore_id, s_ab.logic_name as semaphored_logic_name, s.dependent_job_id as semaphored_job_id,
+   NULL as dependent_semaphore_url, c_ab.logic_name as controlling_logic_name, count(*) as job_count,
```

Comment:
I wonder if you forgot to add `c_j.analysis_id` in the GROUP BY so that it shows the job count for each pair of analysis. Otherwise I'd rename this column to `example_controlling_logic_name`